nbinteract/figure.py

```diff
@@ -0,0 +1,259 @@
+        fig=fig, marks=[bq.Hist], options=options, params=params
+    )
+    plotting._add_marks(fig, [hist])
+
+    def x_fn_wrapped(**interact_params):
+        hist.sample = util.maybe_call(x_fn, interact_params)
+
+    return [x_fn_wrapped]
+
+
+@plotting.use_options([
+    'title', 'aspect_ratio', 'animation_duration', 'xlabel', 'ylabel', 'ylim'
+])
+def _Bar(x_fn, y_fn, *, options={}):
+    params = {
+        'marks': [{
+            'x':
+                plotting._array_or_placeholder(
+                    x_fn, plotting.PLACEHOLDER_RANGE
+                ),
+            'y':
+                plotting._array_or_placeholder(y_fn)
+        }]
+    }
+    fig = options.get('_fig', False) or _create_fig(
+        x_sc=bq.OrdinalScale, options=options
+    )
+    [bar] = plotting._create_marks(
+        fig=fig, marks=[bq.Bars], options=options, params=params
+    )
+    plotting._add_marks(fig, [bar])
+
+    def x_fn_wrapped(**interact_params):
+        x_data = util.maybe_call(x_fn, interact_params, prefix='x')
+        bar.x = x_data
+
+    def y_fn_wrapped(**interact_params):
+        y_data = util.maybe_call(y_fn, interact_params, prefix='y')
+        bar.y = y_data
+
+    return [x_fn_wrapped, y_fn_wrapped]
+
+
+@plotting.use_options([
+    'title', 'aspect_ratio', 'animation_duration', 'xlabel', 'ylabel', 'xlim',
+    'ylim', 'marker'
+])
+def _Scatter(x_fn, y_fn, *, options={}):
+    params = {
+        'marks': [{
+            'x': plotting._array_or_placeholder(x_fn),
+            'y': plotting._array_or_placeholder(y_fn),
+            'marker': plotting._get_option('marker'),
+        }]
+    }
+    fig = options.get('_fig', False) or _create_fig(options=options)
+    [scat] = plotting._create_marks(
+        fig=fig, marks=[bq.Scatter], options=options, params=params
+    )
+    plotting._add_marks(fig, [scat])
+
+    def x_fn_wrapped(**interact_params):
+        x_data = util.maybe_call(x_fn, interact_params, prefix='x')
+        scat.x = x_data
+
+    def y_fn_wrapped(**interact_params):
+        y_data = util.maybe_call(y_fn, interact_params, prefix='y')
+        scat.y = y_data
+
+    return [x_fn_wrapped, y_fn_wrapped]
+
+
+@plotting.use_options([
+    'title', 'aspect_ratio', 'animation_duration', 'xlabel', 'ylabel', 'xlim',
+    'ylim'
+])
+def _Line(x_fn, y_fn, *, options={}):
+    """
+    Internal Line function that generates the Line marker and the wrapper
+    functions for the line.
+    """
+    fig = options.get('_fig', False)
+    [line] = plotting._create_marks(fig=fig, marks=[bq.Lines], options=options)
+    plotting._add_marks(fig, [line])
+
+    def x_fn_wrapped(**interact_params):
+        x_data = util.maybe_call(x_fn, interact_params, prefix='x')
+        line.x = x_data
+
+    def y_fn_wrapped(**interact_params):
+        y_data = util.maybe_call(y_fn, interact_params, prefix='y')
+        line.y = y_data
+
+    return [x_fn_wrapped, y_fn_wrapped]
+
+
+class Plot(object):
```

Comment:
Can we add a docstring for this class and its `__init__` method? According to PEP 257 ([link](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings)):

> The docstring for a class should summarize its behavior and list the public methods and instance variables. If the class is intended to be subclassed, and has an additional interface for subclasses, this interface should be listed separately (in the docstring). The class constructor should be documented in the docstring for its `__init__` method. Individual methods should be documented by their own docstring.

In particular, we should ask ourselves "If I had no idea how to create a Plot, would would I need to know in order to get started quickly?"

Reply:
will do soon, writing iz hard